core/src/main/java/org/geysermc/geyser/skin/FakeHeadProvider.java

```diff
@@ -91,11 +94,38 @@ public SkinProvider.SkinData load(@Nonnull FakeHeadEntry fakeHeadEntry) throws E
                 }
             });
 
-    public static void setHead(GeyserSession session, PlayerEntity entity, CompoundTag profileTag) {
-        SkinManager.GameProfileData gameProfileData = SkinManager.GameProfileData.from(profileTag);
-        if (gameProfileData == null) {
-            return;
+    public static void setHead(GeyserSession session, PlayerEntity entity, CompoundTag NBTData) {
+        if (NBTData.get("SkullOwner") instanceof CompoundTag profileTag) {
```

Comment:
We don't need to call the `get` here twice. The line above here, do `Tag skullOwner = nbtData.get("SkullOwner")` then do the `instanceof` calls below with the `skullOwner` variable.